pip/pip-425.md

```diff
@@ -0,0 +1,127 @@
+# PIP-425: Support connecting with next available endpoint for multi-endpoint serviceUrls
+Implementation PR: [#24387](https://github.com/apache/pulsar/pull/24387)
+
+# Background knowledge
+In the current Pulsar client versions, consumer or producer creation is the responsibility of PulsarClient, we can build a PulsarClient by pass serviceUrl param. The serviceUrl param supports multiple endpoints with the syntax `pulsar://localhost1:6650,localhost2:6650,localhost3:6650,...` or `pulsar+ssl://localhost1:6650,localhost2:6650,localhost3:6650,...`. What happens if some of the endpoints are unavailable? Yes, in the current version, it may cause the consumer or producer to fail to create, which is also the optimization goal of this pip.
```

Comment:
drop "local" prefix in the examples to make them more generic. For pulsar+ssl, the example port should be 6651.

Reply:
Could this be explained more directly? "Currently when an unavailable endpoint is used from the list of multiple endpoint, the client will fail to create a consumer or producer. " And also explaining whether this error is transient or persistent.